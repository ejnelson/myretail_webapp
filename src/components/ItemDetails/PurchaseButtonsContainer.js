import React from 'react';
import './ItemDetails.css';

const PurchaseButtonsContainer = () => (
  <div className="PurchaseButtonsContainer">Purchasebuttons</div>
);

export default PurchaseButtonsContainer;
