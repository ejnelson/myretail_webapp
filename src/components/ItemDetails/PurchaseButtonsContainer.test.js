import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import PurchaseButtonsContainer from './PurchaseButtonsContainer';

describe('PurchaseButtonsContainer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PurchaseButtonsContainer />, div);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<PurchaseButtonsContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  const numbers = { one: '1', two: '2', zero: '0' };
  it('renders the right amount of purchase buttons', () => {
    // Render the PurchaseButtonsContainer
    const container0 = mount(
      <PurchaseButtonsContainer purchasingChannelCode={numbers.zero} />
    );
    const count0 = container0.find('PurchaseButton');
    expect(count0.length).toEqual(2);
    const container1 = mount(
      <PurchaseButtonsContainer purchasingChannelCode={numbers.one} />
    );
    const count1 = container1.find('PurchaseButton');
    expect(count1.length).toEqual(1);
    const container2 = mount(
      <PurchaseButtonsContainer purchasingChannelCode={numbers.two} />
    );
    const count2 = container2.find('PurchaseButton');
    expect(count2.length).toEqual(1);
  });
});
