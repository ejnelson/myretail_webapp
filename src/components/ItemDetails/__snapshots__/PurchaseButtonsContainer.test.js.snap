// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseButtonsContainer renders correctly 1`] = `
<div
  className="purchaseButtonContainer"
>
  <div
    className="buttonContainer"
  >
    <button
      className="purchaseButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "borderColor": "black",
        }
      }
    >
      PICK UP IN STORE
    </button>
    <p
      className="subText"
    >
      find in a store
    </p>
  </div>
  <div
    className="buttonContainer"
  >
    <button
      className="purchaseButton"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(200, 0, 0)",
          "borderColor": "rgb(200, 0, 0)",
        }
      }
    >
      ADD TO CART
    </button>
    <p
      className="subText"
    />
  </div>
</div>
`;
